e2e/keywords/volume.resource

```diff
@@ -1,34 +1,57 @@
 *** Settings ***
-Documentation    Volume Keywords
+Documentation       Longhorn volume related keywords resources
+
+Library             ../libs/keywords/volume_keywords.py
+Resource            ./node.resource
 
-Library    ../libs/keywords/volume_keywords.py
 
 *** Keywords ***
-Create a volume ${size} GB with ${replica_count} replicas
-    ${volume_name} =    create_volume   ${size}    ${replica_count}
-    attach_volume    ${volume_name}
+A ${size} GB ${volume_type} volume with ${number_of_replicas} replicas
+    Given Prepare ${size} GB Volume
+    And Volume ${volume_name} is prepared with ${number_of_replicas} replicas
+    And Volume ${volume_name} is prepared with ${volume_type} access mode
+    And Get the index of the cluster nodes
```

Comment:
I've referenced the [example](https://github.com/longhorn/longhorn-tests/pull/1300#discussion_r1274344624) from the comment to update, and it will be like below:

```robotframework
*** Keywords ***
Create Volume With Fields
    [Arguments]    ${list_of_fields}
    ${volume_name}    ${manifest} =    Create Volume Base Manifest

    FOR    ${field}    IN    @{list_of_fields}
        ${manifest} =    Update Manifest With Fields    ${manifest}    ${field}
    END

    Create Volume With Manifest    ${manifest}

Create Volume Base Manifest
    ${volume_name}    ${manifest} =    create_volume_manifest
    Set Test Variable    ${volume_name}
    RETURN    ${volume_name}    ${manifest}

Update Manifest With Fields
    [Arguments]    ${manifest}    ${field}
    update_manifest_field    ${manifest}    ${field}
    RETURN    ${manifest}

Create Volume With Manifest
    [Arguments]    ${manifest}
    create_volume_using_manifest    ${manifest}

*** Test Cases ***
Node power off with replica-VA
    [Documentation]    volume with replica on attached node and powering off the volume attached node
    [Tags]    nodepoweringoff    3replicas    attachreplicasamenode    abnormalreplicanode    abnormalattachednode
    # Recover: After restart the node, volume should eventually be detached and replica is failed-Need manual to do reattached.
    ${number_of_replicas}=    Convert To Integer    3
    Set Test Variable    ${number_of_replicas}

    ${field1}=    Convert To String    {"spec": {"size": "${${volume_size_gb} * ${Gi}}"}}
    ${field2}=    Convert To String    {"spec": {"numberOfReplicas": ${number_of_replicas}}}
    ${field3}=    Convert To String    {"spec": {"accessMode": "${volume_type}"}}
    @{list_of_fields}=    Create List    ${field1}    ${field2}    ${field3}

    Given Create Volume With Fields    ${list_of_fields}
    And Attach volume to node 1
    And Write data into mount point
```

I also make sure it is executable
![image](https://github.com/longhorn/longhorn-tests/assets/17548901/4394ed89-3755-4cd5-8acb-2143c5cc7a41)

Reply:
@longhorn/qa @innobead This creates a rule about how to write a test case. Not so sure about this. WDYT?